testcases/kernel/syscalls/ioctl/ioctl07.c

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright (c) 2017 Carlo Marcelo Arenas Belon <carenas@gmail.com>
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+/*
+ * Very basic test for the RND* ioctls.
+ *
+ * Reads the entropy available from both /proc and the ioctl and compares
+ * they are similar enough (within a configured fuzz factor)
+ *
+ */
+
+#include <linux/random.h>
+#include <stdlib.h>
+#include "tst_safe_stdio.h"
+#include "tst_test.h"
+
+static char *s_fuzz;
+static int fuzz = 2;
+static struct tst_option options[] = {
+	{"f:", &s_fuzz, "-f c     Fuzz factor to considered valid (default 2)"},
+	{NULL, NULL, NULL}
+};
+
+static int fd;
+
+static void verify_ioctl(void)
+{
+	int cnt, pcnt;
+
+	FILE *proc = SAFE_FOPEN("/proc/sys/kernel/random/entropy_avail", "r");
+
+	SAFE_IOCTL(fd, RNDGETENTCNT, &cnt);
+	if (fscanf(proc, "%d", &pcnt) != 1)
+		tst_res(TBROK | TERRNO, "/proc failure");
```

Comment:
The error message here is a bit confusing. It should be a bit more descriptive something as: "fscanf(/proc/.../entropy_avail, ...) failed".